packages/hub/src/lib/parse-safetensors-metadata.ts

```diff
@@ -14,7 +14,7 @@ export const SAFETENSORS_INDEX_FILE = "model.safetensors.index.json";
 /// but in some situations safetensors weights have different filenames.
 export const RE_SAFETENSORS_FILE = /\.safetensors$/;
 export const RE_SAFETENSORS_INDEX_FILE = /\.safetensors\.index\.json$/;
-export const RE_SAFETENSORS_SHARD_FILE = /\d{5}-of-\d{5}\.safetensors$/;
+export const RE_SAFETENSORS_SHARD_FILE = /(?<shard>\d{5})-of-(?<total>\d{5})\.safetensors$/;
```

Comment:
```suggestion
export const RE_SAFETENSORS_SHARD_FILE = /^(?<prefix>.*)[_-](?<shard>\d{5})-of-(?<total>\d{5})\.safetensors$/;
```

Reply:
```suggestion
export const RE_SAFETENSORS_SHARD_FILE = /^(?<prefix>.*?)[_-]?(?<shard>\d{5})-of-(?<total>\d{5})\.safetensors$/;
```

This is more accurate I think - it makes the `[-_]` optional but still tries to leave it out of `prefix`

Reply:
great idea !

Reply:
but that was my point, the `-_` is *not* optional

Reply:
we do want to enforce some level of convention